### PR TITLE
Facilitate enabling Prometheus stats with a system property

### PR DESCRIPTION
**Pull request by Nirothipan**
Fixes https://github.com/wso2/micro-integrator/issues/147

We can pass the system property **enablePrometheusApi** as

--DenablePrometheusApi to enable prometheus backend.

### CODE REVIEW

**Comment by abeykoon** [comment]
What about using `-DenablePrometheusStats` instread of `-DenablePrometheusBE`?

**Comment by Nirothipan** [comment]
> What about using `-DenablePrometheusStats` instread of `-DenablePrometheusBE`?

+1

**Comment by abeykoon** [comment]
This flag should need to go into documents. 
Do we have a doc for system property arguments we can pass into Micro Integrator? This flag should go into that.

**Comment by Nirothipan** [comment]
> What about using `-DenablePrometheusStats` instread of `-DenablePrometheusBE`?

But we are enabling the ESB to function as Prometheus Back end here. Will it be okay to have it as 
`-DenablePrometheusStats`

**Comment by Nirothipan** [comment]
> This flag should need to go into documents.
> Do we have a doc for system property arguments we can pass into Micro Integrator? This flag should go into that.

It is WIP https://github.com/wso2/micro-integrator/pull/146

File: components/mediation-monitor/mediation-data-publisher/org.wso2.carbon.prometheus.publisher/src/main/java/org/wso2/carbon/prometheus/publisher/service/MetricResource.java [hunk 1]
```diff
@@ -42,11 +38,13 @@
 
     private static Log log = LogFactory.getLog(MetricResource.class);
     private MetricPublisher metricPublisher;
+    private boolean isPrometheusPublishingEnabled;
 
     public MetricResource(String urlTemplate) {
 
         super(urlTemplate);
         metricPublisher = new MetricPublisher();
+        isPrometheusPublishingEnabled = Boolean.parseBoolean(System.getProperty("enablePrometheusBE"));
```
Comment (isudana) [inline components/mediation-monitor/mediation-data-publisher/org.wso2.carbon.prometheus.publisher/src/main/java/org/wso2/carbon/prometheus/publisher/service/MetricResource.java]: shall we rename it to something else? something like enablePrometheusAPI? @a5anka whyt?

Comment (isudana) [reply components/mediation-monitor/mediation-data-publisher/org.wso2.carbon.prometheus.publisher/src/main/java/org/wso2/carbon/prometheus/publisher/service/MetricResource.java]: Also, shall we move the parameter to a constant?

Comment (Nirothipan) [reply components/mediation-monitor/mediation-data-publisher/org.wso2.carbon.prometheus.publisher/src/main/java/org/wso2/carbon/prometheus/publisher/service/MetricResource.java]: Done in https://github.com/wso2/carbon-mediation/pull/1229/commits/ebd74f7c222d652980ff4b3ff59bd478c7d0509e

**Pull request merged**